### PR TITLE
Add Pypac Proxy auto-config for Python

### PR DESCRIPTION
**Pull request by rxm7706**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by rxm7706** [comment]
@conda-forge-admin, please rerender

**Comment by rxm7706** [comment]
@conda-forge/staged-recipes  ready for review

File: recipes/pypac/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,44 @@
+{% set name = "pypac" %}
+{% set version = "0.16.4" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/carsonyl/pypac/archive/refs/tags/v0.16.4.tar.gz
```
Comment (ocefpaf) [inline recipes/pypac/meta.yaml:9]:
```suggestion
  url: https://github.com/carsonyl/pypac/archive/refs/tags/v{{ version }}.tar.gz
```

File: recipes/pypac/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+requirements:
+  host:
+    - python
+    - pip
+  run:
+    - python
+    - requests >=2.0.0
+    - tldextract >=2.2.3
+    - dukpy >=0.2.2
+    - pyobjc-framework-SystemConfiguration >=3.2.1  # [osx]
```
Comment (ocefpaf) [inline recipes/pypac/meta.yaml:25]:
```suggestion
    - pyobjc-framework-SystemConfiguration >=3.2.1  # [osx]
    # Keep the package noarch but allow for the osx only dep
    - __linux  # [linux]
    - __osx    # [osx]
    - __win    # [win]
```

File: recipes/pypac/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  version: {{ version }}
+
+source:
+  url: https://github.com/carsonyl/pypac/archive/refs/tags/v0.16.4.tar.gz
+  sha256: f6eee4f3764568a3c0c401edf60a70485e370dcff4e35d1da503454eb1989e72
+
+build:
+  script: {{ PYTHON }} -m pip install . -vv
+  number: 0
+  skip: true  # [py<37]
```
Comment (ocefpaf) [inline recipes/pypac/meta.yaml:15]:
```suggestion
```

File: recipes/pypac/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+requirements:
+  host:
+    - python
```
Comment (ocefpaf) [inline recipes/pypac/meta.yaml:18]:
```suggestion
    - python >=3.7
```

File: recipes/pypac/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+    - pip
+  run:
+    - python
```
Comment (ocefpaf) [inline recipes/pypac/meta.yaml:21]:
```suggestion
    - python >=3.7
```

**Comment by rxm7706** [comment]
@ocefpaf thanks for showing the virtual packages, the documentation and examples seem sparse. 
Is this how I would make the package work as noarch -- and the linter is expected to fail. 
Thanks!

**Pull request merged**